src/tool/subcommands/archive_cmd.rs

```diff
@@ -607,6 +641,128 @@ async fn merge_snapshots(
+            MultihashCode::Blake2b256.digest(&snap_meta_cbor_encoded),
+        ),
+        data: snap_meta_cbor_encoded,
+    };
+
+    let roots = nunny::vec![snap_meta_block.cid];
+    let snap_meta_frame = {
+        let mut encoder =
+            crate::db::car::forest::new_encoder(DEFAULT_FOREST_CAR_COMPRESSION_LEVEL)?;
+        snap_meta_block.write(&mut encoder)?;
+        anyhow::Ok((
+            vec![snap_meta_block.cid],
+            crate::db::car::forest::finalize_frame(
+                DEFAULT_FOREST_CAR_COMPRESSION_LEVEL,
+                &mut encoder,
+            )?,
+        ))
+    };
+    let f3_frame = {
+        let mut encoder =
+            crate::db::car::forest::new_encoder(DEFAULT_FOREST_CAR_COMPRESSION_LEVEL)?;
+        let f3_data_len = f3_data.seek(SeekFrom::End(0))?;
+        f3_data.seek(SeekFrom::Start(0))?;
+        encoder.write_car_block(f3_cid, f3_data_len as _, &mut f3_data)?;
+        anyhow::Ok((
+            vec![f3_cid],
+            crate::db::car::forest::finalize_frame(
+                DEFAULT_FOREST_CAR_COMPRESSION_LEVEL,
+                &mut encoder,
+            )?,
+        ))
+    };
+
+    let block_frames = crate::db::car::forest::Encoder::compress_stream_default(
+        car_stream.map_err(anyhow::Error::from),
+    );
+    let frames = futures::stream::iter([snap_meta_frame, f3_frame]).chain(block_frames);
+
+    let temp_output = {
+        let mut dir = output.clone();
+        if dir.pop() {
+            tempfile::NamedTempFile::new_in(dir)?
+        } else {
+            tempfile::NamedTempFile::new_in(".")?
+        }
+    };
+    let mut writer = tokio::io::BufWriter::new(tokio::fs::File::create(&temp_output).await?);
+
+    let handle = tokio::spawn({
+        let start = Instant::now();
+        let temp_output = temp_output.path().to_owned();
+        let output = output.clone();
+        async move {
+            let mut interval = tokio::time::interval(tokio::time::Duration::from_secs(1));
+            println!("Getting ready...");
+            loop {
+                interval.tick().await;
+                let snapshot_size = std::fs::metadata(&temp_output)
+                    .map(|meta| meta.len())
+                    .unwrap_or(0);
+                print!(
+                    "{}{}",
+                    anes::MoveCursorToPreviousLine(1),
+                    anes::ClearLine::All
+                );
+                let elapsed_secs = start.elapsed().as_secs_f64();
+                println!(
+                    "{}: {} ({}/s)",
+                    output.display(),
+                    snapshot_size.human_count_bytes(),
+                    if elapsed_secs.is_zero() {
+                        0.
+                    } else {
+                        (snapshot_size as f64) / elapsed_secs
+                    }
+                    .human_count_bytes(),
+                );
+                _ = std::io::stdout().flush();
+            }
+        }
+    });
```

Comment:
@hanabi1224 I think instead of using this whole block we can do this just by using:

```rust=
let pb = ProgressBar::new_spinner().with_style(
        indicatif::ProgressStyle::with_template(
            "{spinner} Merging snapshot... {total_bytes} written in {elapsed} ({binary_bytes_per_sec})",
        )
        .expect("indicatif template must be valid"),
    );
    pb.enable_steady_tick(std::time::Duration::from_millis(100));
    let mut writer = pb.wrap_async_write(writer);
```

Reply:
Fixed.